src/services/anchor-service.ts

```diff
@@ -530,9 +537,15 @@ export default class AnchorService {
       candidateLimit = candidates.length
     }
 
-    for (let i = 0; i < candidateLimit; i++) {
+    for (let i = 0; i < candidates.length; i++) {
       const candidate = candidates[i]
 
+      if (numSelectedCandidates >= candidateLimit) {
```

Comment:
this is an unrelated bug fix I snuck in here.  I can separate it into its own commit if you think it necessary, just did this for expediency.  But the old logic to select unprocessedRequests after the first loop had an error in the logic of where it started processing candidates that causes it to report the wrong number of unprocessedRequests